### PR TITLE
fix some masochistic compiler warnings for the GCC 9 pre-release

### PR DESCRIPTION
**Pull request by andlaus**
the flags which I used are
```
-pedantic \
-Wall \
-Wextra \
-Wformat-nonliteral \
-Wcast-align
-Wpointer-arith \
-Wmissing-declarations \
-Wcast-qual \
-Wshadow
-Wwrite-strings \
-Wchar-subscripts \
-Wredundant-decls \
-fstrict-overflow \
-O3 \
-march=native \
-DNDEBUG=1
```

note that some heavy filtering is not the worst idea because DUNE is far from not emiting any warnings with these flags.

Also, there were some pesky warnings in test_ecl_output which I don't know how to fix:

```
tests/test_ecl_output.cc:218:73: warning: missing initializer for member ‘Opm::data::Connection::effective_Kh’ [-Wmissing-field-initializers]
```

### CODE REVIEW

**Comment by andlaus** [comment]
jenkins build this please

File: ebos/equil/initstateequil.hh [hunk 1]
```diff
@@ -713,9 +713,9 @@ phaseSaturations(const Grid& grid,
                                          /*storeSaturation=*/true,
                                          /*storeDensity=*/false,
                                          /*storeViscosity=*/false,
-                                         /*storeEnthalpy=*/false> SatOnlyFluidState;
+                                         /*storeEnthalpy=*/false> MySatOnlyFluidState;
 
-    SatOnlyFluidState fluidState;
+    MySatOnlyFluidState fluidState;
     typedef typename MaterialLawManager::MaterialLaw MaterialLaw;
```
Comment (totto82) [inline ebos/equil/initstateequil.hh]: What is the reasoning here?

Comment (andlaus) [reply ebos/equil/initstateequil.hh]: naming conflict with https://github.com/OPM/ewoms/blob/master/ebos/equil/equilibrationhelpers.hh#L97-L108

until the EQUIL code gets seriously refactored, this is the best way to chicken out, IMO.

Comment (totto82) [reply ebos/equil/initstateequil.hh]: Ok

**Pull request merged**